### PR TITLE
fix(styles): clean up table sort button styles

### PR DESCRIPTION
**Pull request by anastasialanz**
BREAKING CHANGE: The TableHeader sort button text colors do not change when ascending or descending, per the UX Pin. The removed styles were overriding with the same hex value for light or dark mode.

https://preview.uxpin.com/4052c8b6f05ce03e28813f28deb3a682950c2ac6#/pages/140808126/simulate/sitemap

https://cauldron.dequelabs.com/components/Table

sort-button

### CODE REVIEW

File: packages/styles/table.css [hunk 1]
```diff
@@ -15,7 +14,6 @@
   --table-header-text-color: var(--white);
   --table-header-background-color: var(--accent-dark);
   --table-header-sorting-background-color: #53636e;
-  --table-header-sorting-text-color: var(--white);
```
Comment (anastasialanz) [inline packages/styles/table.css:18]: This is the same color as `--table-header-text-color`

File: packages/styles/table.css [hunk 2]
```diff
@@ -2,7 +2,6 @@
   --table-header-text-color: var(--gray-90);
   --table-header-background-color: var(--gray-20);
   --table-header-sorting-background-color: #caddec;
-  --table-header-sorting-text-color: var(--gray-90);
```
Comment (anastasialanz) [inline packages/styles/table.css:5]: This is the same color as `--table-header-text-color`. The sorting color does not change.

**Review by thuey — COMMENTED** [review]
These changes look good to me. However, it seems like there's still some disagreement about whether or not this should be considered a breaking change. Continuing the discussion elsewhere.

**Comment by anastasialanz** [comment]
Based on https://github.com/dequelabs/cauldron/issues/881, do we agree that this is a breaking change? @dequelabs/cauldron-team

**Comment by thuey** [comment]
> Based on #881, do we agree that this is a breaking change since classes are removed? @dequelabs/cauldron-team

Thoughts, @scurker ?

**Comment by scurker** [comment]
That proposal hasn't yet been discussed/approved so I would lean towards this not being a breaking change.

**Comment by anastasialanz** [comment]
https://github.com/dequelabs/cauldron/blob/develop/CONTRIBUTING.md#styles-1

Based on the breaking changes section for styles: 
> For cauldron's styles library, any of the following would be considered breaking changes:
> - Changing or removing a class name
> - Changing or removing a global css custom property name (see variables.css)
> - Changing or removing a component css custom property name


I think this change would be breaking because I am removing a component's css custom property name.

**Review by scurker — COMMENTED** [review]
These are breaking changes, so per the recommendation of the [contributing guidelines](https://github.com/dequelabs/cauldron/blob/develop/CONTRIBUTING.md#deprecating) we would want to mark these as deprecated, in this instance by including a comment above each deprecated item and warning to not use.

Once the 2 month period deprecation has passed we could remove them in a future PR.

**Comment by anastasialanz** [comment]
> > Is there a certain commit format for adding deprecation comments?
> 
> Not specifically. It might be good to link back to the issue to remove the styles in the comments but I don't have strong opinions here aside from having some sort of comment.

I'm gonna set a reminder in my calendar to remove them.

**Comment by scurker** [comment]
> I'm gonna set a reminder in my calendar to remove them.

We'll also review that ticket tomorrow in our review and stick it in some milestone so it won't be forgotten.

**Pull request merged**